packages/fxa-settings/src/components/BrandMessaging/index.tsx

```diff
@@ -0,0 +1,145 @@
+import { useMetrics } from '../../lib/metrics';
+import { navigate } from '@reach/router';
+import { ReactComponent as IconM } from './icon_m.svg';
+import { ReactComponent as IconClose } from './icon_close.svg';
+import { FtlMsg } from 'fxa-react/lib/utils';
+
+export const bannerClosedLocalStorageKey =
+  '__fxa_storage.fxa_disable_brand_banner';
+
+const BrandMessaging = ({
+  mode,
+  viewName,
+}: {
+  mode?: string;
+  viewName: string;
+}) => {
+  const { brandMessagingMode } = useConfig();
+  const { logViewEventOnce } = useMetrics();
+
+  if (mode === undefined) {
+    mode = brandMessagingMode;
+  }
+
+  const [bannerClosed, setBannerClosed] = useState<string | null>(
+    localStorage.getItem(`${bannerClosedLocalStorageKey}_${mode}`)
+  );
+
+  const bannerVisible =
+    !bannerClosed && (mode === 'prelaunch' || mode === 'postlaunch');
+
+  useEffect(() => {
+    if (bannerVisible) {
+      // hack to ensure banner doesn't overlap, since the only reasonable to way to do this was
+      // with fixed layout.
+      setTimeout(() => {
+        document.body.classList.add('brand-messaging');
+      }, 0);
+    }
+  });
+
+  // Short circuit if mode is unknown or user has previously closed banner
+  if (!bannerVisible) {
+    return <></>;
+  }
+
+  logViewEventOnce(`flow.${viewName}`, `brand-messaging-${mode}-view`);
+
+  function onClickLearnMore() {
+    logViewEventOnce(`flow.${viewName}`, `brand-messaging-${mode}-learn-more`);
+    navigate(
+      'https://support.mozilla.org/kb/firefox-accounts-renamed-mozilla-accounts'
+    );
+  }
+
+  function onClickCloseBanner() {
+    logViewEventOnce(
+      `flow.${viewName}`,
+      `brand-messaging-${mode}-banner-close`
+    );
+    document.body.classList.remove('brand-messaging');
+    localStorage.setItem(`${bannerClosedLocalStorageKey}_${mode}`, 'true');
+    setBannerClosed('true');
+  }
+
+  return (
+    <>
+      <div id="banner-brand-message" className="fixed w-full top-0 left-0">
+        <div className="flex relative justify-center p-2 brand-banner-bg">
+          {mode === 'prelaunch' && (
+            <div className="flex" data-testid="brand-prelaunch">
+              <div className="m-logo flex-none relative mt-1 mr-4 mb-2">
+                <div>
+                  <IconM />
+                </div>
+              </div>
+              <div className="flex-initial max-w-md pr-2">
+                <p className="text-left text-sm font-bold">
+                  <FtlMsg id="brand-prelaunch-title">
+                    Firefox accounts will be renamed Mozilla accounts on Nov 1
+                  </FtlMsg>
+                </p>
+                <p className="text-left text-xs">
+                  <FtlMsg id="brand-prelaunch-subtitle">
+                    You’ll still sign in with the same username and password,
+                    and there are no other changes to the products that you use.
+                  </FtlMsg>
+                  <span
```

Comment:
Better to use the native `a` element for a11y and then prevent.default it. Otherwise add `role="link"`

Reply:
Alright, I had an `a` initially, but we want to record a metric on click, so I switched it to this this approach. 

Do you know what happens if I have the `a` with an href, and an onclick that sends a metric. Will the metric get recorded? It seemed a bit unclear to me. If I put the on click on the `a` tag, and leave off the href, I'll get that warning about href being required...

Reply:
Doing href ='#', still results in a warning iirc. I might just do the role='link'. This feels cleaner than the preventDefault in some ways. I think there are a couple this types of situations in our code, so revisiting make sounds good to me.